cEP-0016.md

```diff
@@ -0,0 +1,234 @@
+`GitCommitBear`.
+
+## Introduction
+
+The `GitCommitBear` at present performs a check on the content of regular
+commits made to coala. However, there are many special types of commit
+messages that should be used only in conjunction with patches containing a
+special type of content.
+
+For example, `git revert` commit which is used to revert any previous commit and
+`git merge` commit which is generated whenever someone performs a GitHub rebase
+and tries to merge master onto the pull request. Rebase and merge on GitHub will
+always update the committer information and create new commit SHAs which is not
+correct. These commits generate a default commit message that is very concise
+and not according to coala’s standards. The coala project wants to allow
+`git revert` commits, but does not want to accept `git merge` commits as the
+repositories are strictly fast-forward merges that do not have an extra commit.
+
+Also in cases where some tests are not required by any pull request made to
+coala, some special sequences can be added to git commit messages to instruct
+tools to operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in `.coafile`
+
+## Background
+
+Below are some code snippets from `.coafile` of some projects that actively use
+`GitCommitBear`.
+
+```
+[commits]
+bears = `GitCommitBear`
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+shortlog_trailing_period = False
+```
+
+```
+[all.commit]
+bears = `GitCommitBear`
+shortlog_trailing_period = False
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+body_close_issue = True
+body_close_issue_full_url = True
+body_close_issue_on_last_line = True
+body_enforce_issue_reference = False
+```
+
+```
+[commit]
+bears = `GitCommitBear`
+```
+
+## Proposed Change
+
+This project would involve making three meta bears that would identify
+commits that skip CI buids, `git merge` and `git revert` commits.
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely Travis
+   merge commits, `git revert` , `git merge` and commits that disable CI builds.
+2. Each type of commit is then handled as a separate case.
+3. For Travis commits:
+   Travis commits are unseen merge commits that are sent to CI when GitHub merge
+   commits are added to 3+ neatly stacked PRs. These merge commits are different
+   as they are not made using `git merge`. Instead they are made by Github
+   automatically when a PR is commited by more than two people. `GitCommitBear`
+   at present does not identify such commits and thus fails. This failure is
+   then reflected in Travis builds causing it to fail as well.
+   - As a solution to this problem, settings will be added in `GitCommitBear`
+     that would identify Travis merge commits and ignore them. Therefore, the
+     `GitCommitBear` will run without failing and so will the Travis build.
```

Comment:
Cool I was only concerned about this part :)
Also I can now see more clearly how are you gonna handle it. I will try to review that one quickly
https://github.com/coala/coala-bears/pull/2501
And let's head over there for further discussion.
@jayvdb Can this be merged ? :) It's a green signal from my side.

Reply:
ok, this part is confusing to me. If I didn't know how github handles merges, I'd be totally lost. This part is not about the `proposed solution`, but is more related to the problem statement, but it definitely need a rephrasing, and I'd rather avoid the use of *Travis merge*, but instead simply say *merges generated automatically by Github*.